### DISCUSSION TITLE
How to setup a secure code evaluator in LearnR like rstudio.cloud does?

### DISCUSSION THREAD

**Discussion by Elvincth** [rstudio/learnr#703] [Q&A] [answered]
When we use the  self-hosted version of Learn R or hosted version in  shinyapps.io, the user can print out variables that contain the tutorial source code using the code evaluator. 

For example in  [shinyapps](https://learnr-examples.shinyapps.io/ex-data-filter/#section-filter-rows-with-filter) hosted version, the user can run `.server_context` to get the source code of the current rmd:

[![enter image description here][1]][1]

**(UPDATED) Example 2:**

By running `cat(system("cat /srv/connect/apps/ex-data-filter/ex-data-filter.Rmd", intern = TRUE), sep = '\n')` can get the source code of the current rmd too:
[![enter image description here][2]][2]

But interestingly in [rstudio.cloud](https://rstudio.cloud/learn/primers/3.4) hosted version, the code evaluator seems to be sandboxed, and all "sensitive" variables cannot be run through the evaluator.

[![enter image description here][3]][3]

Can I get some advice on how to implement such a secure code evaluator?

Thanks!

  [1]: https://i.stack.imgur.com/hEWOK.jpg
  [2]: https://i.stack.imgur.com/ZrB09.jpg
  [3]: https://i.imgur.com/5oOknhx.jpeg

**Comment by gadenbuie** [comment, marked as answer]
The short answer is that, specifically in shinyapps.io, it's not really possible to block access to that file. 😞
